File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.py
def solution(A,B):
    A.sort()
    B.sort(reverse=True)
    #B.sort()
    #B=B[::-1]
    print(f'A : {A}')
    print(f'B : {B}')
    
    for i in range(len(A)):
        A[i]=A[i]*B[i]
    
    answer = 0
    for i in A:
        print(i)
        answer+=i


    return answer

File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.java
import java.util.*;

class Solution
{
    public int solution(int []A, int []B)
    {
        int ans=0;
        Arrays.sort(A);
        Arrays.sort(B);
        //Arrays.sort(B, Collections.reverseOrder()); //내림차순
        
        for(int i=0;i<A.length;i++){
            ans+=A[i]*B[B.length-1-i];  //B의 큰 값부터 곱하기
        }
        
        return ans;
    }
}